src/rgw/rgw_reshard.cc

```diff
@@ -0,0 +1,185 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rgw_rados.h"
+#include "rgw_reshard.h"
+#include "cls/rgw/cls_rgw_client.h"
+#include "cls/lock/cls_lock_client.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+const string reshard_oid = "reshard";
+const string reshard_lock_name = "reshard_process";
+
+RGWReshard::RGWReshard(CephContext *_cct, RGWRados* _store):cct(_cct), store(_store)
+{
+    max_jobs = cct->_conf->rgw_reshard_max_jobs;
+}
+
+int RGWReshard::get_io_ctx(librados::IoCtx& io_ctx)
+{
+  string reshard_pool = store->get_zone_params().reshard_pool.name;
```

Comment:
@oritwas should not access reshard_pool.name, won't work with namespaces. Just take a reference to `rgw_pool`